src/mwparserfromhell/wikicode.py

```diff
@@ -516,19 +516,43 @@ def matches(self, other):
         adjusted. Specifically, whitespace and markup is stripped and the first
         letter's case is normalized. Typical usage is
         ``if template.name.matches("stub"): ...``.
+
+        If either side has any colons, everything before the last colon is taken to be
+        a namespace and/or interwiki prefix.  The parts before and after the colon are
+        normalized and compared separately; both must match for the result to be True.
```

Comment:
This does not make sense - if something is *not a valid namespace/interwiki prefix*, it should not be normalized.